src/Route.php

```diff
@@ -4,109 +4,119 @@
 
 namespace Yiisoft\Router;
 
+use Attribute;
 use InvalidArgumentException;
 use RuntimeException;
 use Stringable;
 use Yiisoft\Http\Method;
-use Yiisoft\Middleware\Dispatcher\MiddlewareDispatcher;
 
 use function in_array;
 
 /**
  * Route defines a mapping from URL to callback / name and vice versa.
  */
-final class Route implements Stringable
+#[Attribute(Attribute::TARGET_METHOD | Attribute::TARGET_CLASS | Attribute::IS_REPEATABLE)]
+class Route implements Stringable
 {
-    private ?string $name = null;
-
+    private bool $actionAdded = false;
     /**
-     * @var string[]
+     * @var array[]|callable[]|string[]
      */
-    private array $hosts = [];
-    private bool $override = false;
-    private bool $actionAdded = false;
-
+    private array $builtMiddlewares = [];
     /**
      * @var array[]|callable[]|string[]
      */
-    private array $middlewareDefinitions = [];
-
-    private array $disabledMiddlewareDefinitions = [];
-
+    private array $middlewares = [];
```

Comment:
Keep previous name `middlewareDefinitions`. It more accurately reflects the contents of the property.